.github/workflows/iroha2-deploy-base-image.yml

```diff
@@ -0,0 +1,81 @@
+name: I2::Base Image Deploy
+
+on:
+  push:
+    branches: 
+      - iroha2-dev
+      - iroha2
+      - iroha2-lts
+    paths:
+      - 'Dockerfile.base'
+
+env:
+  REPOSITORY_URL: '${{ github.server_url }}/${{ github.repository }}.git'
+
+jobs:
+  base_image_deploy_iroha2-dev:
+    if: github.ref == 'refs/heads/iroha2-dev'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up buildx
+        uses: docker/setup-buildx-action@v2
+        id: buildx
+        with:
+          install: true
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push iroha2-ci:dev image
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: hyperledger/iroha2-ci:dev
+          context: ${{ env.REPOSITORY_URL }}#iroha2-dev
+          file: Dockerfile.base
+
+  base_image_deploy_iroha2:
+    if: github.ref == 'refs/heads/iroha2'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up buildx
+        uses: docker/setup-buildx-action@v2
+        id: buildx
+        with:
+          install: true
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push iroha2-ci:stable image
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: hyperledger/iroha2-ci:stable
+          context: ${{ env.REPOSITORY_URL }}#iroha2
+          file: Dockerfile.base
+
+  base_image_deploy_iroha2-lts:
+    if: github.ref == 'refs/heads/iroha2-lts'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up buildx
+        uses: docker/setup-buildx-action@v2
+        id: buildx
+        with:
+          install: true
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push iroha2-ci:lts image
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: hyperledger/iroha2-ci:lts
+          context: ${{ env.REPOSITORY_URL }}#iroha2-lts
+          file: Dockerfile.base
```

Comment:
```suggestion
jobs:
  base_image_deploy:
    strategy:
      matrix:
        tags: [dev, stable, lts]
    if: github.ref == 'refs/heads/iroha2-${{ matrix.tags }}'
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Set up buildx
        uses: docker/setup-buildx-action@v2
        id: buildx
        with:
          install: true
      - name: Login to DockerHub
        uses: docker/login-action@v2
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
      - name: Build and push iroha2-ci:${{ matrix.tags }} image
        uses: docker/build-push-action@v3
        with:
          push: true
          tags: hyperledger/iroha2-ci:${{ matrix.tags }}
          context: ${{ env.REPOSITORY_URL }}#iroha2-${{ matrix.tags }}
          file: Dockerfile.base
          build-args: TAG=${{ matrix.tags }}
```

To apply this, we need to unify `iroha2` and `iroha2-stable` with `iroha2-stable`. What do you think @appetrosyan

Reply:
Technically the `stable` `lts` and `dev` are all tags, not branches. Unification might be a good idea.

Reply:
Let me make a clarification. I'm suggesting to synchronize tags and branches.

![image](https://user-images.githubusercontent.com/49983831/182807091-6d7e3659-15a2-423a-8d50-70e6948c4c30.png)
![image](https://user-images.githubusercontent.com/49983831/182807252-1c51101a-6d23-478b-b880-617f3014dc40.png)

So these branches will be `iroha2-stable`

Reply:
I think this is a question of which tag we prefer, implicit `latest` or explicit `stable`

Reply:
I'll rename `iroha2` to `iroha2-stable` branch if so